src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SuggestionHandlers/CallNodeSuggestionHandler.cs

```diff
@@ -59,34 +59,28 @@ internal override bool TryAddSuggestionsForNodeKind(IntellisenseData.Intellisens
                     intellisenseData.BoundTo = intellisenseData.Binding.ErrorContainer.HasErrors(callNode) ? string.Empty : callNode.Head.Name;
                     IntellisenseHelper.AddSuggestionsForFunctions(intellisenseData);
                 }
-                else if (callNode.Token.Span.Lim > cursorPos || callNode.ParenClose == null)
+                else if (callNode.Token.Span.Lim > cursorPos)
                 {
                     // Handling the erroneous case when user enters a space after functionName and cursor is after space.
                     // Cursor is before the open paren of the function.
                     // Eg: "Filter | (" AND "Filter | (some Table, some predicate)"
                     return false;
                 }
-                else
+                else if (callNode.ParenClose != null && cursorPos <= callNode.ParenClose.Span.Min)
                 {
-                    // If there was no closed parenthesis we would have an error node.
-                    Contracts.Assert(callNode.ParenClose != null);
-
-                    if (cursorPos <= callNode.ParenClose.Span.Min)
-                    {
-                        // Cursor position is before the closed parenthesis and there are no arguments.
-                        // If there were arguments FindNode should have returned one of those.
-                        if (intellisenseData.CurFunc != null && intellisenseData.CurFunc.MaxArity > 0)
-                        {
-                            IntellisenseHelper.AddSuggestionsForTopLevel(intellisenseData, callNode);
-                        }
-                    }
-                    else if (IntellisenseHelper.CanSuggestAfterValue(cursorPos, intellisenseData.Script))
+                    // Cursor position is before the closed parenthesis and there are no arguments.
+                    // If there were arguments FindNode should have returned one of those.
+                    if (intellisenseData.CurFunc != null && intellisenseData.CurFunc.MaxArity > 0)
                     {
-                        // Verify that cursor is after a space after the closed parenthesis and
-                        // suggest binary operators.
-                        IntellisenseHelper.AddSuggestionsForAfterValue(intellisenseData, intellisenseData.Binding.GetType(callNode));
+                        IntellisenseHelper.AddSuggestionsForTopLevel(intellisenseData, callNode);
                     }
                 }
+                else if (IntellisenseHelper.CanSuggestAfterValue(cursorPos, intellisenseData.Script))
```

Comment:
>lse if ( [](http://example.com/codeflow?start=17&length=8)

Should there be a condition here ?
if(callNode.ParenClose == null && IntellisenseHelper.CanSuggestAfterValue(cursorPos, intellisenseData.Script))

Could following be possibility that cursor is beyond closed paran?

Filter(...) |

Reply:
good catch, Thanks.